### PR TITLE
Fixing not being able to create comments on local community posts.

### LINKED ISSUE #4853: [Bug] Local-only community comments cannot be acted upon (0.19.4)
### Requirements

- [X] This is a bug report, and if not, please post to https://lemmy.ml/c/lemmy_support instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this issue already exists.
- [X] It's a single bug. Do not report multiple bugs in one issue.
- [X] It's a frontend issue, not a backend issue; Otherwise please create an issue on the [backend repo](https://github.com/LemmyNet/lemmy) instead.

### Summary

We created a new local-only community as a test on our instance and it is currently impossible to upvote, downvote or delete/remove a comment in a post made in this community.

Any such action causes the `Couldn't find comment.` toast notification to show up.

On the mobile app I use (Sync for Lemmy), the comments aren't showing up in that thread.

### Steps to Reproduce

1. Create local-only community
2. Submit a post
3. Create a comment
4. Try to upvote or downvote (error)
5. Delete the comment (error)
6. (as mod or admin) Remove a comment with a reason (error)

### Technical Details

Happens on all platforms I could test on (Firefox for desktop and mobile).
image

Screenshot_20240614-105823

### Lemmy Instance Version

0.19.4

### Lemmy Instance URL

https://lemmy.ca/c/local

### PR DESCRIPTION
**Pull request by dessalines**
- This was caused by not passing my_person_id into various `CommentView::read` functions. The `visible_communities_only` function only allows you to view local communities if my_person is passed.
- Fixes #4853

@dullbananas logically that check doesn't make sense to me. Should the visible_communities_only function be checking for a `local_user_id`, since those are local only, not a `person_id`, which could be a federated person?

### CODE REVIEW

**Review by dullbananas — APPROVED** [review]
> Should the visible_communities_only function be checking for a local_user_id, since those are local only, not a person_id, which could be a federated person?

That function would work incorrectly for a remote person, but it would be incorrect usage of the function. Accidentally doing this could be prevented by using `&LocalUser` instead of `PersonId` in visible_communities_only and all of the read functions, just like the list functions.

**Comment by Nutomic** [comment]
Yes checking with `LocalUserId` would make more sense. I dont think its a problem in practice because remote users cannot call the api, and I dont think this check is used for federation. Still better to change it.

**Review by Nutomic — APPROVED** [review]

**Comment by dessalines** [comment]
Okay this is done. I needed to refactor several views to receive the `LocalUser`, rather than the `PersonId`.

File: crates/apub/src/activities/create_or_update/comment.rs [hunk 1]
```diff
@@ -178,8 +179,10 @@ impl ActivityHandler for CreateOrUpdateNote {
     // anyway.
     // TODO: for compatibility with other projects, it would be much better to read this from cc or
     // tags
-    let mentions = scrape_text_for_mentions(&comment.content);
-    send_local_notifs(mentions, comment.id, &actor, do_send_email, context).await?;
+    if let Some(local_user) = LocalUserView::read_person(&mut context.pool(), actor.id).await? {
```
Comment (dessalines) [inline crates/apub/src/activities/create_or_update/comment.rs:182]: This check probably should've been done already, but now it makes sure that the local user exists before sending notifs.

**Review by Nutomic — APPROVED** [review]

File: crates/api_common/src/build_response.rs [hunk 2]
```diff
@@ -80,14 +80,9 @@ pub async fn build_post_response(
...
     .is_ok();
-  let post_view = PostView::read(
-    &mut context.pool(),
-    post_id,
-    Some(person.id),
-    is_mod_or_admin,
-  )
-  .await?
-  .ok_or(LemmyErrorType::CouldntFindPost)?;
+  let post_view = PostView::read(&mut context.pool(), post_id, None, is_mod_or_admin)
```
Comment (dullbananas) [inline crates/api_common/src/build_response.rs:83]: Using `None` here will cause problems, such as making the post like API return an error for local-only communities. `person: &Person` should be changed to `local_user: &LocalUser`.

**Review by dullbananas — CHANGES_REQUESTED** [review]

File: crates/api_common/src/build_response.rs (hunk 2 above)
Comment (dessalines) [reply crates/api_common/src/build_response.rs]: Good catch, updated. Hopefully tests will pass now.

**Review by dullbananas — APPROVED** [review]

**Review by Nutomic — APPROVED** [review]

**Pull request merged**